lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -505,6 +507,69 @@ def add_marketplace_data(post, options)
         end
       end
 
+      def add_level_two_three_data(post, options)
+        post[:processing] ||= {}
+        post[:customer] ||= {}
+
+        # American Express only supports Level 2 data.
+        # Only is required add items info in lvl2 data for amex
+        add_items(post, options)
+        add_level_two_data(post, options)
+        add_level_three_data(post, options)
+        add_shipping_data(post, options)
+      end
+
+      def add_items(post, options)
+        items = build_items(options[:line_items])
+        post[:items] = items unless items.nil?
+      end
+
+      def add_level_two_data(post, options)
+        post[:customer][:tax_number] = options[:tax_number] # field no require for amex
+
+        post[:processing].merge!(
+          {
+            order_id: options.dig(:processing, :order_id),
+            tax_amount: options[:tax_amount]
+          }.compact
+        )
+      end
+
+      def add_level_three_data(post, options)
+        post[:processing].merge!(
+          {
+            discount_amount: options[:discount_amount],
+            shipping_amount: options[:shipping_amount],
+            duty_amount: options[:duty_amount]
+          }.compact
+        )
+      end
+
+      def add_shipping_data(post, options)
+        post[:shipping] ||= {}
+        post[:shipping][:from_address_zip] = options[:from_address_zip]
+      end
+
+      def build_items(line_items)
```

Comment:
💬 Comment:

Just an idea => if you set line_items with a default value `[]` you will not need the really return